### ISSUE TITLE
v1.4.1 - broken build

### ISSUE THREAD

**Issue by jbulow** [BrickSchema/Brick#654]
Following the instructions and trying to build v1.4.1 results in:

```
 make test
mkdir -p extensions
python tools/sort_definitions.py bricksrc/definitions.csv
python generate_brick.py
Creating new OntoEnv
Traceback (most recent call last):
  File "/big/Brick/generate_brick.py", line 22, in <module>
    from bricksrc.env import env
  File "/big/Brick/bricksrc/env.py", line 3, in <module>
    env = OntoEnv(cfg)
          ^^^^^^^^^^^^
ValueError: IO error for operation on rec/Source/SHACL/RealEstateCore: No such file or directory (os error 2)
```

**Comment by gtfierro** [comment]
RealEstateCore is now a git submodule on Brick, so you'll need to `git submodule init` and `git submodule update` to get that directory to appear! (Note to self: I'll need to adjust the documentation for this)

**Comment by jbulow** [comment]
After adding the rec submodule and running "make test" I get a lot of ERROR-logging like:

```
Creating new OntoEnv
[2024-08-27T19:38:35Z ERROR ontoenv] Failed to read ontology File("examples/ebu3b_brick.ttl"): error while parsing IRI 'http://ucsd.edu/building/ontology/ebu3b#EBU3B_RM_3xxx_3rd_FLR_SUPFLO_AVG_%': Invalid IRI percent encoding '%' on line 10977 at position 77
[2024-08-27T19:38:39Z ERROR ontoenv] Failed to read ontology File("examples/gtc_brick.ttl"): error while parsing IRI 'https://brickschema.org/schema/1.0.2/building_example#building:gtc/district_hx/port[po]': Invalid IRI code point '[' on line 48 at position 90
[2024-08-27T19:38:42Z ERROR ontoenv] Failed to read ontology File("examples/ghc_brick.ttl"): error while parsing IRI 'http://cmu.edu/building/ontology/ghc#SCSC_Hillman/4400_S_Corridor/RH_Valve_%': Invalid IRI percent encoding '%' on line 364 at position 79
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:22Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:23Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:23Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology Url("http://data.ashrae.org/standard223/1.0/extensions/settings"): Failed to fetch ontology from http://data.ashrae.org/standard223/1.0/extensions/settings
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology file http://data.ashrae.org/standard223/1.0/extensions/settings: Failed to fetch ontology from http://data.ashrae.org/standard223/1.0/extensions/settings
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:24Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:25Z ERROR ontoenv] Failed to read ontology Url("https://brickschema.org/schema/1.4/Brick"): error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
[2024-08-27T19:39:25Z ERROR ontoenv] Failed to read ontology file https://brickschema.org/schema/1.4/Brick: error while parsing IRI '?xml version="1.0" encoding="utf-8"?': Invalid IRI code point ' ' on line 1 at position 39
...
```

Is that expected?

**Comment by gtfierro** [comment]
Yes, they are! I know that's confusing, and sorry for not having the new tooling adequately documented.

We use a tool called [OntoEnv](https://github.com/gtfierro/ontoenv-rs) to manage the ontology dependencies. We have it configured to act in "offline" mode here so that we can vendor all the dependencies in the repository. Some of the files here refer to Brick 1.4, but they can't (and shouldn't) read it. There is some bug here, because ontoenv shouldn't be complaining about not reaching Brick 1.4. I've made a note to look into that soon.

tl;dr there is a bug preventing it from reading Brick 1.4, but this is "okay" for now because it actually *shouldn't* be trying to read Brick 1.4

Brick should still build/test correctly. I'm verifying this myself right now

**Comment by gtfierro** [comment]
@jbulow if you update `pyontoenv==0.1.9` it should fix those error messages

**Comment by gtfierro** [comment]
@jbulow can you confirm if the build is fixed for you?

**Comment by jbulow** [comment]
Still the same problems after updating pyontenv to 0.1.9. Is there some kind of cleanup needed?

**Comment by gtfierro** [comment]
Hi @jbulow You might need to remove the `.ontoenv` directory for this to work? I've tried this in a fresh repo with `pyontoenv==0.1.9` and I see this output (below).

There are still 3 errors from the examples/ folder, but these do not affect the build.

```
(Brick) ➜  Brick git:(master) make
mkdir -p extensions
python tools/sort_definitions.py bricksrc/definitions.csv
python generate_brick.py
No module named 'rdflib_sqlalchemy'
2024-09-12:09:39:41,495 WARNING [__init__.py:22] sqlalchemy not installed. SQL-backed graph support will not be available. Try 'pip install brickschema[persistence]' to install it.
Creating new OntoEnv
[2024-09-12T15:39:42Z ERROR ontoenv] Failed to read ontology File("examples/gtc_brick.ttl"): error while parsing IRI 'https://brickschema.org/schema/1.0.2/building_example#building:gtc/district_hx/port[po]': Invalid IRI code point '[' on line 48 at position 90
[2024-09-12T15:39:42Z ERROR ontoenv] Failed to read ontology File("examples/ghc_brick.ttl"): error while parsing IRI 'http://cmu.edu/building/ontology/ghc#SCSC_Hillman/4400_S_Corridor/RH_Valve_%': Invalid IRI percent encoding '%' on line 364 at position 79
[2024-09-12T15:39:43Z ERROR ontoenv] Failed to read ontology File("examples/ebu3b_brick.ttl"): error while parsing IRI 'http://ucsd.edu/building/ontology/ebu3b#EBU3B_RM_3xxx_3rd_FLR_SUPFLO_AVG_%': Invalid IRI percent encoding '%' on line 10977 at position 77
Running ['/bin/sh', '/private/tmp/Brick/.venv/lib/python3.10/site-packages/brickschema/topquadrant_shacl/bin/shaclinfer.sh'] -datafile /var/folders/n8/yxqgdlwn4s18f6ncs3srq9gm0000gn/T/tmp04n0ajrh/data.ttl
Got 0 inferred triples
2024-09-12:09:39:59,453 INFO    [generate_brick.py:842] Beginning BRICK Ontology compilation
2024-09-12:09:39:59,453 INFO    [generate_brick.py:846] Inheriting annotations down the subclass trees
2024-09-12:09:39:59,456 INFO    [generate_brick.py:878] Defining properties
2024-09-12:09:39:59,459 INFO    [generate_brick.py:897] Defining Point subclasses
2024-09-12:09:39:59,519 WARNING [generate_brick.py:317] Property 'tags' not defined for https://brickschema.org/schema/Brick#Differential_Setpoint
2024-09-12:09:39:59,569 WARNING [generate_brick.py:317] Property 'tags' not defined for https://brickschema.org/schema/Brick#Velocity_Pressure_Sensor
2024-09-12:09:39:59,574 WARNING [generate_brick.py:317] Property 'tags' not defined for https://brickschema.org/schema/Brick#Refrigerant_Level_Sensor
2024-09-12:09:39:59,895 INFO    [generate_brick.py:912] Defining Equipment, System and Location subclasses
2024-09-12:09:39:59,940 WARNING [generate_brick.py:317] Property 'tags' not defined for https://brickschema.org/schema/Brick#Lighting
2024-09-12:09:40:00,002 INFO    [generate_brick.py:923] Defining Measurable hierarchy
2024-09-12:09:40:00,024 INFO    [generate_brick.py:988] Adding applicable units
2024-09-12:09:40:00,120 INFO    [generate_brick.py:1010] Defining entity properties
2024-09-12:09:40:00,221 INFO    [generate_brick.py:1022] Adding deprecations
2024-09-12:09:40:00,225 INFO    [generate_brick.py:1028] Adding class definitions
2024-09-12:09:40:00,243 INFO    [generate_brick.py:759] Inferred setpoint: Close_Setpoint
2024-09-12:09:40:00,245 WARNING [generate_brick.py:771] WARNING: Close_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Close_Limit.
2024-09-12:09:40:00,319 INFO    [generate_brick.py:759] Inferred setpoint: Current_Setpoint
2024-09-12:09:40:00,321 WARNING [generate_brick.py:771] WARNING: Current_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Current_Limit.
2024-09-12:09:40:00,321 INFO    [generate_brick.py:759] Inferred setpoint: Position_Setpoint
2024-09-12:09:40:00,323 WARNING [generate_brick.py:771] WARNING: Position_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Position_Limit.
2024-09-12:09:40:00,324 WARNING [generate_brick.py:771] WARNING: Position_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Max_Position_Setpoint_Limit.
2024-09-12:09:40:00,326 WARNING [generate_brick.py:771] WARNING: Position_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Min_Position_Setpoint_Limit.
2024-09-12:09:40:00,337 WARNING [generate_brick.py:771] WARNING: Fresh_Air_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Fresh_Air_Setpoint_Limit.
2024-09-12:09:40:00,339 WARNING [generate_brick.py:771] WARNING: Fresh_Air_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Min_Fresh_Air_Setpoint_Limit.
2024-09-12:09:40:00,340 WARNING [generate_brick.py:771] WARNING: Fresh_Air_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Max_Fresh_Air_Setpoint_Limit.
2024-09-12:09:40:00,340 INFO    [generate_brick.py:759] Inferred setpoint: Ventilation_Air_Flow_Ratio_Setpoint
2024-09-12:09:40:00,343 WARNING [generate_brick.py:771] WARNING: Ventilation_Air_Flow_Ratio_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#Ventilation_Air_Flow_Ratio_Limit.
2024-09-12:09:40:00,356 WARNING [generate_brick.py:771] WARNING: High_Static_Pressure_Cutout_Setpoint does not exist in Brick for https://brickschema.org/schema/Brick#High_Static_Pressure_Cutout_Setpoint_Limit.
2024-09-12:09:40:00,363 INFO    [generate_brick.py:1031] Adding other .ttl files
2024-09-12:09:40:00,402 INFO    [generate_brick.py:1042] Cleaning up ontology prefixes
2024-09-12:09:40:00,402 INFO    [generate_brick.py:1087] Brick ontology compilation finished! Generated 51202 triples
```

**Comment by jbulow** [comment]
Works fine now. I get the same output as you now when I build from master branch.
